crates/ruff/src/args.rs

```diff
@@ -924,20 +921,15 @@ The path `{value}` does not point to a configuration file"
     }
 }
 
+#[allow(deprecated)]
 fn resolve_output_format(
     output_format: Option<OutputFormat>,
-    preview: bool,
-) -> Option<OutputFormat> {
-    Some(match output_format {
-        Some(o) => o,
-        None => return None
-    }).map(|format| match format {
-        OutputFormat::Text => {
-            warn_user!("`--output-format=text` is deprecated. Use `--output-format=full` or `--output-format=concise` instead. `text` will be treated as `{}`.", OutputFormat::default(preview));
-            OutputFormat::default(preview)
-        },
-        other => other
-    })
+) -> anyhow::Result<Option<OutputFormat>> {
+    if let Some(OutputFormat::Text) = output_format {
+        Err(anyhow!("`--output-format=text` is deprecated. Use `--output-format=full` or `--output-format=concise` instead."))
```

Comment:
We may want to use a word other than "deprecated" here, since that usually implies "available but discouraged", while this is just a hard error. What about: `--output-format=text is no longer supported`?